src/file/FileController.ts

```diff
@@ -0,0 +1,108 @@
+import {File} from './../common/File';
+import {IFileVersion} from './../common/IFileVersion';
+
+export class FileController {
+  /**
+   * Dependencies we want passed to the constructor.
+   * @see http://docs.angularjs.org/guide/di
+   */
+  public static $inject = [
+    '$rootScope'
+  ];
+
+  /**
+   * File object. Passed in as component binding.
+   */
+  public file: File;
+
+  /**
+   * Whether the parent bucket is versioned. Passed in as component binding.
+   */
+  public versioned: boolean;
+
+  /**
+   * File versions.
+   */
+  public versions: IFileVersion[];
+
+  /**
+   * Flag used to indicate a background operation is running.
+   */
+  public working: boolean;
+
+  /**
+   * Show file versions.
+   */
+  public showVersions: boolean;
+
+  /**
+   * Create a file controller instance.
+   */
+  constructor(private $rootScope: ng.IScope) {
+    this.versions = [];
+    this.working = false;
+    this.showVersions = false;
+  }
+
+  /**
+   * Toggle showing file versions.
+   */
+  public toggleVersions() {
+    // hide displayed file versions
+    if (this.showVersions) {
+      this.showVersions = false;
+      return;
+    }
+
+    // load and display file versions
+    this.working = true;
+    this.onLoadVersions()
+      .then((versions: IFileVersion[]) => {
+        this.versions = versions.sort(this.compareFileVersions);
+        this.showVersions = true;
+      })
+      .catch(() => {
+        this.versions = [];
+        this.showVersions = false;
+      })
+      .then(() => {
+        this.working = false;
+        this.$rootScope.$digest();
+      });
+  }
+
+  /**
+   * Load file versions. This is overriden by the component binding.
+   */
+  public onLoadVersions(): Promise<IFileVersion[]> {
+    // overriden by binding but TypeScript requires we return something
+    return new Promise(function (
```

Comment:
Here is where you don't use fat arrow notation, but since you don't reference 'this' in the function there should be no difference.